Update the LHESource from fix in 75.

parbol opened the pull request:
This pull request contains the update that was done in 75 to solve the issue where only 3 files were being read by LHESource.

parbol committed:
Update the LHESource from fix in 75.

cmsbuild commented:
A new Pull Request was created by @parbol  for CMSSW_7_1_X.

Update the LHESource from fix in 75.

It involves the following packages:

GeneratorInterface/LHEInterface


@vciulli, @cmsbuild, @covarell, @bendavid, @thuer can you please review it and eventually sign? Thanks.
@mkirsano this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

bendavid commented:
@parbol the change doesn't look quite the same as the one in https://github.com/cms-sw/cmssw/pull/7795.
In particular there seems to be an extra 
"partonLevel = reader->next(&newFileOpened);"

Did I get it wrong?  What is the reason for the difference?

parbol committed:
There was a problem when merging and I added a line that was not supposed to be there. Fixed now.

parbol commented:
Hi Josh,

The reason for this catch is that just before pushing I changed branch and that line got merged without me noticing. It's obviously a mistake. I have deleted it and updated the code, it should be fine now.

cmsbuild commented:
Pull request #9448 was updated. @vciulli, @cmsbuild, @covarell, @bendavid, @thuer can you please check and sign again.

bendavid commented on GeneratorInterface/LHEInterface/plugins/LHESource.cc:
```diff
@@ -64,8 +64,19 @@ void LHESource::nextEvent()
 
-	bool newFileOpened = false;
-	partonLevel = reader->next(&newFileOpened);
+	//bool newFileOpened = false;
```
please just remove these two lines rather than commenting them.

parbol committed:
Remove commented lines

cmsbuild commented:
Pull request #9448 was updated. @vciulli, @cmsbuild, @covarell, @bendavid, @thuer can you please check and sign again.

bendavid commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_1_X IBs unless changes or unless it breaks tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: b52beebeb3292e674259df8b4f05c1703f86f374
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9448/5301/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_1_X IBs unless changes (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

davidlange6 commented:
+1

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9448/5301/summary.html

There are some workflows for which there are errors in the baseline:
 1001.0 step 2
1003.0 step 2
25202.0 step 2
50202.0 step 2
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

ferencek commented:
The PR description suggests there was a problem with reading more than 3 LHE files in a single LHESource but from my tests in CMSSW_7_1_16_patch1 the following configuration (LHE files contain 50k events each)
```python
process.maxEvents = cms.untracked.PSet(
    input = cms.untracked.int32(100)
)

# Input source
process.source = cms.Source("LHESource",
    fileNames = cms.untracked.vstring(
        'root://cmsxrootd.fnal.gov//store/user/ferencek/LHE/Stop2ToStop1H_13TeV-madgraph/stop2_1000.0_0_decayed_1000006_200__1000022_25.lhe',
        'root://cmsxrootd.fnal.gov//store/user/ferencek/LHE/Stop2ToStop1H_13TeV-madgraph/stop2_1000.0_0_decayed_1000006_400__1000022_225.lhe'
    ),
    skipEvents = cms.untracked.uint32(50100)
)
```
would result in a fatal exception
```
----- Begin Fatal Exception 05-Jun-2015 12:17:12 CDT-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing run: 1
   [1] Calling beginRun for unscheduled module Pythia8HadronizerFilter/'generator'
Exception Message:
Principal::getByLabel: Found zero products matching all criteria
Looking for type: LHERunInfoProduct
Looking for module label: source
Looking for productInstanceName: 
Looking for process: LHEFile
----- End Fatal Exception -------------------------------------------------
```
In CMSSW_7_1_16_patch2, after this PR got included, it all runs as expected. So it seems there was already a problem with reading more than 1 LHE file. Can an expert please comment?

Also note that the above exception causes CRAB jobs to fail in case they attempt to read events beyond the first file. Interestingly enough, if a job started reading events at the end of the first file and continued to the second, that worked just fine.
